### PR TITLE
Xcode SDK build fixes

### PR DESCRIPTION
**Pull request by pepone**
This PR includes a fix to build C++/Objetive-C xcodesdk on macOS arm64
There is also a fix for Util.py swift testing and a other for IceUtil/stacktrace testing

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -123,3 +123,4 @@ Carthage
 .classpath
 .project
 .settings
+*.dep
```
Comment (externl) [inline .gitignore:126]: Where do these `*.dep` come from?

File: scripts/ice-builder-xcode-wrapper [hunk 2]
```diff
@@ -15,4 +15,14 @@ else
     FLAGS="--ice-home ../../../.."
 fi
 
-/usr/local/bin/icebuilder $FLAGS "$@"
+if [ -f "/usr/local/bin/icebuilder" ]; then
```
Comment (externl) [inline scripts/ice-builder-xcode-wrapper:18]: Should we first check the path?

**Review by externl — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (pepone) [reply .gitignore:126]: I saw those after opening and build the controller projects with Xcode

File: scripts/ice-builder-xcode-wrapper (hunk 2 above)
Comment (pepone) [reply scripts/ice-builder-xcode-wrapper:18]: How will you do that?

Comment (externl) [reply scripts/ice-builder-xcode-wrapper:18]: Something like `if command -v icebuilder >/dev/null 2>&1; then` should work I think. Or we could maybe user `which ic builder`

Comment (pepone) [reply scripts/ice-builder-xcode-wrapper:18]: Ok I going to merge it as is, we can update it later if required

**Pull request merged**